tests/conftest.py

```diff
@@ -651,3 +662,190 @@ async def testagent(
     finally:
         p.terminate()
         p.wait()
+
+
+# OTLP Constants
+PROTOBUF_HEADERS = {"Content-Type": "application/x-protobuf"}
+JSON_HEADERS = {"Content-Type": "application/json"}
+
+
+# OTLP Helper Functions
+async def _get_http_status_from_metadata(call):
+    """Extract HTTP status from GRPC trailing metadata."""
+    trailing_metadata = await call.trailing_metadata()
+    for key, value in trailing_metadata:
+        if key == "http-status":
+            return int(value)
+    return None
+
+
+# Common OTLP Test Data Fixtures
+@pytest.fixture
+def service_name():
+    return "ddservice"
+
+
+@pytest.fixture
+def environment():
+    return "ddenv"
+
+
+@pytest.fixture
+def version():
+    return "ddv1"
+
+
+@pytest.fixture
+def host_name():
+    return "ddhost"
+
+
+# OTLP Infrastructure Fixtures
+@pytest.fixture
+def otlp_http_url(testagent_url):
+    parsed_url = urlparse(testagent_url)
+    return f"{parsed_url.scheme}://{parsed_url.hostname}:{DEFAULT_OTLP_HTTP_PORT}"
+
+
+@pytest.fixture
+def otlp_test_client(otlp_http_url):
+    """OTLP client for retrieving stored logs, metrics and requests."""
+    parsed_url = urlparse(otlp_http_url)
+    client = TestOTLPClient(parsed_url.hostname, parsed_url.port, parsed_url.scheme)
+    client.clear()
+    client.wait_to_start()
+    yield client
+    client.clear()
+
+
+@pytest.fixture(params=["logs", "metrics"])
+async def otlp_grpc_client(request):
+    """GRPC client that can connect to either logs or metrics service."""
+    channel = grpc_aio.insecure_channel(f"127.0.0.1:{DEFAULT_OTLP_GRPC_PORT}")
+
+    if request.param == "logs":
+        stub = LogsServiceStub(channel)
+    elif request.param == "metrics":
+        stub = MetricsServiceStub(channel)
+    else:
+        raise ValueError(f"Unknown service type: {request.param}")
+
+    yield stub
+
+    await channel.close()
+
+
+@pytest.fixture
+async def otlp_logs_grpc_client():
+    """GRPC client specifically for logs service."""
+    channel = grpc_aio.insecure_channel(f"127.0.0.1:{DEFAULT_OTLP_GRPC_PORT}")
+    stub = LogsServiceStub(channel)
+
+    yield stub
+
+    await channel.close()
+
+
+@pytest.fixture
+async def otlp_metrics_grpc_client():
+    """GRPC client specifically for metrics service."""
+    channel = grpc_aio.insecure_channel(f"127.0.0.1:{DEFAULT_OTLP_GRPC_PORT}")
+    stub = MetricsServiceStub(channel)
+
+    yield stub
+
+    await channel.close()
+
+
+@pytest.fixture
+async def grpc_server_with_failure_type(agent_app, available_port, aiohttp_server, request):
```

Comment:
I don't understand the difference between `grpc_server_with_failure_type` and `service_grpc_server_with_failure_type`. Do we need both?